flink-runtime/src/main/java/org/apache/flink/runtime/state/DefaultOperatorStateBackend.java

```diff
@@ -148,21 +170,27 @@ public void close() throws IOException {
 	@Override
 	public void dispose() {
 		IOUtils.closeQuietly(closeStreamOnCancelRegistry);
-		registeredStates.clear();
+		registeredOperatorStates.clear();
+		registeredBroadcastStates.clear();
 	}
 
 	// -------------------------------------------------------------------------------------------
 	//  State access methods
 	// -------------------------------------------------------------------------------------------
 
+	@Override
+	public <K, V> ReadWriteBroadcastState<K, V> getBroadcastState(MapStateDescriptor<K, V> stateDescriptor) throws Exception {
```

Comment:
I think we don't need this because the other `getBroadcastState()` is only ever called with `BROADCAST` mode.